lib/bolt/bolt_option_parser.rb

```diff
@@ -6,6 +6,16 @@
 
 module Bolt
   class BoltOptionParser < OptionParser
+    EXCLUDES = { 'apply' => %w[params],
+                 'command' => %w[execute tmpdir boltdir modulepath compile-concurrency params],
+                 'file' => %w[execute boltdir modulepath compile-concurrency params],
+                 'plan' => %w[execute],
+                 'puppetfile' => %w[nodes targets query rerun user params password private-key host-key-check
+                                    ssl ssl-verify run-as sudo-password concurrency compile-concurrency modulepath
+                                    boltdir inventoryfile save-rerun trannsport connect-timeout tty tmpdir execute],
+                 'script' => %w[execute boltdir modulepath compile-concurrency params],
+                 'task' => %w[execute compile-concurrency] }.freeze
```

Comment:
Not sure if this is the best approach to hide options from the help text, but it feels cleaner than writing separate methods for each command to toggle the hide variable. This lets you just add an option that should be hidden to a command's array.